src/shared/generator.c

```diff
@@ -370,54 +370,63 @@ int generator_write_fsck_deps(
 int generator_write_timeouts(
                 const char *dir,
                 const char *what,
-                const char *where,
                 const char *opts,
-                char **filtered) {
+                char **ret_filtered) {
 
-        /* Configure how long we wait for a device that backs a mount point or a
-         * swap partition to show up. This is useful to support endless device timeouts
-         * for devices that show up only after user input, like crypto devices. */
+        /* Configure how long we wait for a device that backs a mount point or a swap partition to
+         * show up. This is useful to support endless device timeouts for devices that show up only
+         * after user input, like crypto devices.
+         *
+         * Note that we set *ret_filtered even on failure, so that the calling code can ignore the
+         * failure for the snippet if it wishes, but continue with the other options.
+         */
 
         _cleanup_free_ char *node = NULL, *unit = NULL, *timeout = NULL;
         usec_t u;
         int r;
 
+        assert(what);
+
         r = fstab_filter_options(opts, "comment=systemd.device-timeout\0"
                                        "x-systemd.device-timeout\0",
-                                 NULL, &timeout, NULL, filtered);
+                                 NULL, &timeout, NULL, ret_filtered);
+        if (r == -ENOMEM)
+                return log_oom();
         if (r < 0) {
-                log_warning_errno(r, "Failed to parse fstab options, ignoring: %m");
-                return 0;
+                log_warning_errno(r, "%s: failed to parse fstab options, ignoring: %m", what);
+                *ret_filtered = NULL;
```

Comment:
Otherwise the initialization becomes quite weird here: we might initialize this when it's OOM and caller considers it fatal